### ISSUE TITLE
Video upload without required creator should not be possible

### ISSUE THREAD

**Issue by abias** [Opencast-Moodle/moodle-block_opencast#185]
As an admin, I have configured the metadata fields of this block in a way that the title field and the creator field is mandatory:
![grafik](https://user-images.githubusercontent.com/1092118/117438404-ce0aba80-af31-11eb-9aa3-92c9df7204f0.png)

While the requirement of the title field works fine, teachers can still upload videos without any creator as the requirement is not enforced when the upload form is submitted.

This feels like a bug of the autocompletion field type.

**Comment by TamaraGunkel** [comment]
Hi,
I think that's an issue with the autocompletion type in Moodle itself. Quickform submits a pseudo value which is successfully validated but later correctly turned into an empty array. I'll raise an issue in Moodle because it should be fixed there. 
Though, I'm still going to add a validation in the block because I guess it takes a while until this is fixed.

**Comment by TamaraGunkel** [comment]
That the Moodle issue: https://tracker.moodle.org/browse/MDL-71831

I added the validation on client side because it required only minor modifications. I think we can neglect that the autocomplete fields are not validated on server side until this is fixed in the Moodle core. There won't be many people trying to circumvent this and it's always possible to find out who uploaded the video.

**Comment by abias** [comment]
Hi @TamaraGunkel ,

thank you for having a look into this issue and for implementing a solution as well as for the bug report for Moodle core.

As discussed 1:1, I am afraid that the fix does not work for me on Moodle 3.9.7+.
It would be awesome if you could have another look or try to verify that the fix does work for you on 3.9.x as well.
I will be happy to provide more details about my installation / configuration of you need it.

Thanks,
Alex

**Comment by TamaraGunkel** [comment]
Hey,
it's weird that it doesn't work for you. I've looked into the responsible JavaScript Code. It's generated in the RuleRegistry and wasn't changed since 2006. So it should be included in 3.9.x too. 
Could you check if the following snippet is included in the source code and hit when submitting? 

```  
for (var i = 0; i < frm.elements['creator[]'].options.length; i++) {
    if (frm.elements['creator[]'].options[i].selected) {
      value[valueIdx++] = frm.elements['creator[]'].options[i].value;
    }
}

if (value == '' && !errFlag['creator']) {
    errFlag['creator'] = true;
    _qfMsg = _qfMsg + '\n - Required';
}
```
